WebKitBrowser/WebKitImplementation.cpp

```diff
@@ -2869,11 +2869,28 @@ static GSourceFuncs _handlerIntervention =
                      nullptr);
                 g_free(indexedDBPath);
 #else
+#if WEBKIT_CHECK_VERSION(2, 38, 0)
+                uint64_t indexedDBSizeBytes = 0;    // No limit by default, use WebKit defaults (1G at the moment of writing)
+                if (_config.OriginStorageRatio.IsSet() && _config.OriginStorageRatio.Value() != 0) {
+                    Core::File persistentPath(_service->PersistentPath());
+                    indexedDBSizeBytes = ((persistentPath.Core::File::Size() * _config.OriginStorageRatio.Value()) / 100);
```

Comment:
Hi, @VeithMetro @modeveci
"persistentPath.Core::File::Size()" Does it truely return available disk space? With R4.2 it just returns directory size itself (without its contents) which is 4096 in all cases. Sources look similar in R4, master and other branches and there is just `stat()` call inside Thunder/Source/core/FileSystem.cpp:
```
#ifdef __POSIX__
        struct stat data;
        if (stat(_name.c_str(), &data) == 0) {
            _size = data.st_size;
```
On the other hand Core::Partition from the same FileSystem.h file correctly returns partition Size() and Free() space

Reply:
@asurdej-comcast You are right. I thought File::Size would do the same result but apparently not.
On the otherhand, Core::Partition uses statvfs which returns partition size when you passed a directory/file in the partition. It should serve the purpose. 
@VeithMetro Sorry, my misunderstading for File::Size, Can we use partition api?